flexbe_core/src/flexbe_core/logger.py

```diff
@@ -34,6 +38,23 @@ def log(text, severity):
         # also log locally
         Logger.local(text, severity)
 
+    @staticmethod
+    def log_throttle(period, text, severity):
+        # creat unique identifier for each logging message
+        log_id = str(severity)  + "_" + text
+        # only log when it's the first time or period time has passed for the logging message
+        if not log_id in Logger._last_logged.keys() or \
+            rospy.Time.now().to_sec() - Logger._last_logged[log_id].to_sec() > period:
+                Logger.log(text, severity)
+                Logger._last_logged.update({log_id: rospy.Time.now()})
+
+        if len(Logger._last_logged) > Logger.MAX_LAST_LOGGED_SIZE:
```

Comment:
Once we hit limit this will process every message following.

Should we consider throwing away the oldest half of messages, otherwise i fear this limit will become a limiting factor given need to sort and pop when adding a single new message.

Reply:
You're right, although this would only occur under the circumstance that we have more than 1024 `log_throttle`s and it would only happen when a changing message is logged. 
But I agree, that we don't want to have that overhead. I'll change the clearing to a percentage to the `MAX_LAST_LOGGED_SIZE` :+1: